core/vm/interpreter.go

```diff
@@ -209,6 +209,9 @@ func (in *EVMInterpreter) Run(contract *Contract, input []byte, readOnly bool) (
 			var dynamicCost uint64
 			dynamicCost, err = operation.dynamicGas(in.evm, contract, stack, mem, memorySize)
 			cost += dynamicCost // for tracing
+			if err == ErrGasUintOverflow {
```

Comment:
The origin error of dynamicGas is indeed replaced by `ErrOutOfGas`, which I think is not good.

```diff
diff --git a/core/vm/interpreter.go b/core/vm/interpreter.go
index 0ab520b90f..afe6988f47 100644
--- a/core/vm/interpreter.go
+++ b/core/vm/interpreter.go
@@ -208,8 +208,11 @@ func (in *EVMInterpreter) Run(contract *Contract, input []byte, readOnly bool) (
                        // cost is explicitly set so that the capture state defer method can get the proper cost
                        var dynamicCost uint64
                        dynamicCost, err = operation.dynamicGas(in.evm, contract, stack, mem, memorySize)
+                       if err != nil {
+                               return nil, err
+                       }
                        cost += dynamicCost // for tracing
-                       if err != nil || !contract.UseGas(dynamicCost) {
+                       if !contract.UseGas(dynamicCost) {
                                return nil, ErrOutOfGas
                        }
                        // Do tracing before memory expansion
```

Perhaps this change is cleaner?

Reply:
Yes. Thanks.
I fixed to return the all internal error (not only `ErrGasUintOverflow`) in [c707abd](https://github.com/ethereum/go-ethereum/pull/26877/commits/c707abd3441b51f77c2b7b05a2b599411506b4a7).

Reply:
Will try to fix the failed tests.